src/livestreamer/plugins/ruv.py

```diff
@@ -0,0 +1,170 @@
+                                        ras1|
+                                        ras-1|
+                                        ras2|
+                                        ras-2
+                                    )
+                                    /
+                                    [a-zA-Z0-9_-]+
+                                    /
+                                    [0-9]+
+                                    /?
+                                    """, re.VERBOSE)
+
+_rtmp_url_re = re.compile(r"""rtmp://sipvodfs\.fplive.net/sipvod/
+                                (?P<status>
+                                    lokad|
+                                    opid
+                                )
+                                /
+                                (?P<date>[0-9]+/[0-9][0-9]/[0-9][0-9]/)?
+                                (?P<id>[A-Z0-9\$_]+)
+                                \.
+                                (?P<ext>
+                                    mp4|
+                                    mp3
+                                )""", re.VERBOSE)
+
+_id_map = {
+    "ruv": "ruv",
+    "ras1": "ras1",
+    "ras-1": "ras1",
+    "ras2": "ras2",
+    "ras-2": "ras2",
+    "rondo": "ras3"
+}
+
+
+class Ruv(Plugin):
+    @classmethod
+    def can_handle_url(cls, url):
+        if _live_url_re.match(url):
+            return _live_url_re.match(url)
+        else:
+            return _sarpurinn_url_re.match(url)
+
+    def __init__(self, url):
+        Plugin.__init__(self, url)
+        live_match = _live_url_re.match(url)
+
+        if live_match:
+            self.live = True
+            self.channel_path = live_match.group("channel_path")
+        else:
+            self.live = False
+
+    def _get_live_streams(self):
+        stream_id = _id_map[self.channel_path]
+
+        streams = {}
+
+        if stream_id == "ruv":
+            qualities_rtmp = ["720p", "480p", "360p", "240p"]
+
+            for i, quality in enumerate(qualities_rtmp):
+                streams[quality] = RTMPStream(
+                    self.session,
+                    {
+                        "rtmp": RTMP_LIVE_URL.format(stream_id, i+1),
+                        "pageUrl": self.url,
+                        "live": True
+                    }
+                )
+
+            qualities_hls = ["240p_hls", "360p_hls", "480p_hls", "720p_hls"]
+            for i, quality_hls in enumerate(qualities_hls):
+                streams[quality_hls] = HLSStream(
+                    self.session,
+                    HLS_RUV_LIVE_URL.format(i+1)
+                )
+
+        else:
+            streams["audio"] = RTMPStream(self.session, {
+                "rtmp": RTMP_LIVE_URL.format(stream_id, 1),
+                "pageUrl": self.url,
+                "live": True
+            })
+
+            streams["audio_hls"] = HLSStream(
+                self.session,
+                HLS_RADIO_LIVE_URL.format(stream_id)
+            )
+
```

Comment:
To add multiple stream types for the same quality you should either add them as a list in the dict, e.g. `streams["audio"] = [rtmp_stream, hls_stream]` or the more slick way is to return a generator instead of a dict, so you can do `yield "audio", stream` and Livestreamer will automatically add suffixes as needed.